tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestNormalizer.scala

```diff
@@ -31,36 +31,268 @@ class TestNormalizer extends FunSuite with BeforeAndAfterEach {
+    assert(expected == output)
+  }
+
+  test("""~ (x >= y) ~~> x < y""") {
+    val input =
+      tla.not(tla.ge(tla.name("x") as IntT1(), tla.name("y") as IntT1()) as BoolT1()) as BoolT1()
+    val output = normalizer.apply(input)
+    val expected =
+      tla.lt(tla.name("x") as IntT1(), tla.name("y") as IntT1()) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""~(x /= y) ~~> x = y""") {
+    val input =
+      tla.not(tla.neql(tla.name("x") as IntT1(), tla.name("y") as IntT1()) as BoolT1()) as BoolT1()
+    val output = normalizer.apply(input)
+    val expected =
+      tla.eql(tla.name("x") as IntT1(), tla.name("y") as IntT1()) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""~(~FALSE \in s) ~~> ~(TRUE \in s)""") {
+    val input =
+      tla.not(tla.in(tla.not(tla.bool(false)) as BoolT1(), tla.name("s") as SetT1(BoolT1())) as BoolT1()) as BoolT1()
+    val output = normalizer.apply(input)
+    val expected =
+      tla.not(tla.in(tla.bool(true), tla.name("s") as SetT1(BoolT1())) as BoolT1()) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""(~FALSE \notin s) ~~> ~(TRUE \in s)""") {
+    val input =
+      tla.notin(tla.not(tla.bool(false)) as BoolT1(), tla.name("s") as SetT1(BoolT1())) as BoolT1()
+    val output = normalizer.apply(input)
+    val expected =
+      tla.not(tla.in(tla.bool(true), tla.name("s") as SetT1(BoolT1())) as BoolT1()) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""lab(a) :: ~FALSE ~~> lab(a) :: TRUE""") {
+    val input = tla.label(tla.not(tla.bool(false)) as BoolT1(), "lab", "a") as BoolT1()
+    val output = normalizer.apply(input)
+    val expected = tla.label(tla.bool(true), "lab", "a") as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""~<>x ~~> []~x""") {
+    val input =
+      tla.not(tla.diamond(tla.name("x") as BoolT1()) as BoolT1()) as BoolT1()
+    val output = normalizer.apply(input)
+    val expected =
+      tla.box(tla.not(tla.name("x") as BoolT1()) as BoolT1()) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""~[]x ~~> <>~x""") {
+    val input =
+      tla.not(tla.box(tla.name("x") as BoolT1()) as BoolT1()) as BoolT1()
+    val output = normalizer.apply(input)
+    val expected =
+      tla.diamond(tla.not(tla.name("x") as BoolT1()) as BoolT1()) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""LET x = ~FALSE IN ~~x ~~> LET x = TRUE IN x""") {
+    val decl = tla.declOp("x", tla.not(tla.bool(false)) as BoolT1()) as BoolT1()
+    val input = tla.letIn(tla.not(tla.not(tla.name("x") as BoolT1()) as BoolT1()) as BoolT1(), decl) as BoolT1()
+    val output = normalizer.apply(input)
+
+    val expectedDecl = tla.declOp("x", tla.bool(true)) as BoolT1()
+    val expected =
+      tla.letIn(tla.name("x") as BoolT1(), expectedDecl) as BoolT1()
+    assert(expected == output)
+  }
+
+  test("""~ (LET x = ~FALSE IN ~x ~~> LET x = ~FALSE IN x)""") {
```

Comment:
I noticed that we are not normalizing declarations on negated LET IN expressions, but I think we should. Is this intended somehow?

Reply:
Yes! Especially that the comment is saying that we are normalizing the declaration body.